packages/libs/error-stack/src/serde/hook.rs

```diff
@@ -0,0 +1,360 @@
+use alloc::{boxed::Box, vec::Vec};
+use core::{any::TypeId, iter::FusedIterator};
+
+pub(crate) use default::install_builtin_hooks;
+
+use crate::Frame;
+
+pub struct Serde {}
+
+pub type HookContext<T> = crate::hook::context::HookContext<Serde, T>;
+
+fn serialize<'a, T: serde::Serialize + Send + Sync + 'static>(
+    frame: &'a Frame,
+) -> Option<Box<dyn erased_serde::Serialize + 'a>> {
+    #[cfg(nightly)]
+    let value: &T = frame.request_ref()?;
+    #[cfg(not(nightly))]
+    let value: &T = frame.downcast_ref()?;
+    Some(Box::new(value))
+}
+
+// Thanks to https://users.rust-lang.org/t/hrtb-on-multiple-generics/34255/2 for the solution
+pub trait DynamicFn<'a, I>: Send + Sync + 'static
```

Comment:
`Dynamic` does not really say anything about the implied meaning. Maybe we should rename it to something like `SerializeFn`?

Reply:
fixed in [0941b84](https://github.com/hashintel/hash/pull/1558/commits/0941b84cbc9939518d3f5ff1e6fabf6e89a0ea83)